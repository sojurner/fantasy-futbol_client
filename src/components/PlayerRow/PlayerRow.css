tr {
  cursor: pointer;
  color: white;
  background: gray;
  text-align: center;
  font-weight: 700; }

tr:hover {
  color: black;
  background-color: #c1af98; }
