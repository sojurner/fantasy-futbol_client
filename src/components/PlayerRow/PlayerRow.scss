tr {
  cursor: pointer;
  color: white;
  background: gray;
  text-align: center;
  font-weight: 700;
}

tr:hover {
  color: black;
  background-color: rgb(193, 175, 152);
}
