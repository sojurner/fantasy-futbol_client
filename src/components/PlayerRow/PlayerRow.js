import React from 'react';
import './PlayerRow.css';

export const PlayerRow = ({
  id,
  Photo,
  Name,
  Nationality,
  Positions,
  Club,
  Overall,
  Value,
  Wage,
  onOpenModal
}) => {
  return (
    <tr data-id={id} className={Name} onClick={onOpenModal.bind(null, id)}>
      <td>
        <img
          src={Photo}
          onError={e => {
            e.target.onerror = null;
            e.target.src =
              'https://www.referee.com/wp-content/uploads/2012/10/FIFA.png';
          }}
          height="35"
          className="photo"
          alt="alt_text"
        />
      </td>
      <td>{Name}</td>
      <td>{Nationality}</td>
      <td>{Positions}</td>
      <td>{Club}</td>
      <td>{Overall}</td>
      <td>{Value}</td>
    </tr>
  );
};
