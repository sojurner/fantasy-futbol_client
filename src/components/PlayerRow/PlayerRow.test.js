import React from 'react';
import { PlayerRow } from './PlayerRow';

describe('PlayerRow', () => {
  let mockId = 1;
  let mockPhoto = 'https://cdn.sofifa.org/players/4/18/158023.png';
  let mockName = 'DeGea';
  let mockNationality = 'Brazil';
  let mockPositions = 'RW';
  let mockClub = 'Chelsea';
  let mockOverall = 93;
  let mockValue = '€105M';
  let mockWage = '€105M';
  let mockOpenModal = jest.fn();

  let wrapper = shallow(
    <PlayerRow
      id={mockId}
      Photo={mockPhoto}
      Name={mockName}
      Nationality={mockNationality}
      Club={mockClub}
      Overall={mockOverall}
      Value={mockValue}
      Wage={mockWage}
      onOpenModal={mockOpenModal}
    />
  );

  it('should Match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
