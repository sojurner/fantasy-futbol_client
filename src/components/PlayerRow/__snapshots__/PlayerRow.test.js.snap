// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerRow should Match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    Club="Chelsea"
    Name="DeGea"
    Nationality="Brazil"
    Overall={93}
    Photo="https://cdn.sofifa.org/players/4/18/158023.png"
    Value="€105M"
    Wage="€105M"
    id={1}
    onOpenModal={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <td>
          <img
                    alt="alt_text"
                    className="photo"
                    height="35"
                    src="https://cdn.sofifa.org/players/4/18/158023.png"
          />
</td>,
        <td>
          DeGea
</td>,
        <td>
          Brazil
</td>,
        <td />,
        <td>
          Chelsea
</td>,
        <td>
          93
</td>,
        <td>
          €105M
</td>,
        <td>
          €105M
</td>,
      ],
      "className": "DeGea",
      "data-id": 1,
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="alt_text"
            className="photo"
            height="35"
            src="https://cdn.sofifa.org/players/4/18/158023.png"
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "alt_text",
            "className": "photo",
            "height": "35",
            "src": "https://cdn.sofifa.org/players/4/18/158023.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "DeGea",
        },
        "ref": null,
        "rendered": "DeGea",
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Brazil",
        },
        "ref": null,
        "rendered": "Brazil",
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Chelsea",
        },
        "ref": null,
        "rendered": "Chelsea",
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 93,
        },
        "ref": null,
        "rendered": 93,
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "€105M",
        },
        "ref": null,
        "rendered": "€105M",
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "€105M",
        },
        "ref": null,
        "rendered": "€105M",
        "type": "td",
      },
    ],
    "type": "tr",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <td>
            <img
                        alt="alt_text"
                        className="photo"
                        height="35"
                        src="https://cdn.sofifa.org/players/4/18/158023.png"
            />
</td>,
          <td>
            DeGea
</td>,
          <td>
            Brazil
</td>,
          <td />,
          <td>
            Chelsea
</td>,
          <td>
            93
</td>,
          <td>
            €105M
</td>,
          <td>
            €105M
</td>,
        ],
        "className": "DeGea",
        "data-id": 1,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="alt_text"
              className="photo"
              height="35"
              src="https://cdn.sofifa.org/players/4/18/158023.png"
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "alt_text",
              "className": "photo",
              "height": "35",
              "src": "https://cdn.sofifa.org/players/4/18/158023.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "DeGea",
          },
          "ref": null,
          "rendered": "DeGea",
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Brazil",
          },
          "ref": null,
          "rendered": "Brazil",
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Chelsea",
          },
          "ref": null,
          "rendered": "Chelsea",
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 93,
          },
          "ref": null,
          "rendered": 93,
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "€105M",
          },
          "ref": null,
          "rendered": "€105M",
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "€105M",
          },
          "ref": null,
          "rendered": "€105M",
          "type": "td",
        },
      ],
      "type": "tr",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
